src/methods/abstract.js

```diff
@@ -186,7 +187,11 @@ export function AbstractRelationalComparison(
 // ECMA262 7.2.13
 export function AbstractEqualityComparison(realm: Realm, x: ConcreteValue, y: ConcreteValue): boolean {
   // 1. If Type(x) is the same as Type(y), then
-  if (x.getType() === y.getType()) {
+  if (
```

Comment:
This is not very reasonable. Is there any suggestion?

Reply:
We are not hugely concerned about the runtime performance of this operation, so it would be reasonable to factor out this logic into a function, such as "HasSameType".